components/count-badge/count-badge-icon.js

```diff
@@ -75,6 +75,18 @@ class CountBadgeIcon extends CountBadgeMixin(LitElement) {
 		this._badgeId = getUniqueId();
 	}
 
+	connectedCallback() {
+		super.connectedCallback();
+		// re-render when dir attribute is added to reposition for RTL
+		this._mutationObserver = new MutationObserver(() => this.requestUpdate());
+		this._mutationObserver.observe(this, { attributeFilter: ['dir'] });
```

Comment:
AttributeFilter ensures we will only request the update if the dir is changed (which should only happen once). This is valid syntax for all our supported browsers (https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe).